net/netmon/defaultroute_darwin.go

```diff
@@ -83,26 +81,17 @@ func defaultRoute() (d DefaultRouteDetails, err error) {
 		}
 	}
 
-	// Start of our fallback logic if Swift didn't give us an interface name, or gave us an invalid
-	// one.
-	// We start by attempting to use the Wi-Fi interface, which on iPhone is always called en0.
-	enZeroIf := getInterfaceByName("en0")
-	if enZeroIf != nil {
-		log.Println("defaultroute_ios: using en0 (fallback)")
-		d.InterfaceName = enZeroIf.Name
-		d.InterfaceIndex = enZeroIf.Index
-		return d, nil
-	}
+	// Fallback to the BSD logic
 
-	// Did it not work? Let's try with Cellular (pdp_ip0).
-	cellIf := getInterfaceByName("pdp_ip0")
-	if cellIf != nil {
-		log.Println("defaultroute_ios: using pdp_ip0 (fallback)")
-		d.InterfaceName = cellIf.Name
-		d.InterfaceIndex = cellIf.Index
-		return d, nil
+	idx, err := DefaultRouteInterfaceIndex()
+	if err != nil {
+		return d, err
 	}
-
-	log.Println("defaultroute_ios: no running interfaces available")
-	return d, ErrNoGatewayIndexFound
+	iface, err := net.InterfaceByIndex(idx)
```

Comment:
@agottardo This seemed prudent (esp on macOS).   This should give us a better answer, though this code is never actually hit unless something is going terribly wrong already.